### ISSUE TITLE
Error message for JSC_XMODULE_REQUIRE_ERROR is confusing

### ISSUE THREAD

**Issue by dbarnett** [google/closure-compiler#1069]
The `JSC_XMODULE_REQUIRE_ERROR` message looks like
  `namespace "{0}" provided in module {1} but required in module {2}`

It's really hard to parse this message. I think the main point of the error is that the namespace isn't provided in module {2}, and the fact that it _is_ provided in module {1} is just a hint to help with troubleshooting. Could it be changed to something like
  `required "{0}" namespace never provided in module {2} (but provided in module {1})`
?

**Comment by concavelenz** [comment]
It should says effectively: "it is in {1} and {2} doesn't depend on {1}.

**Comment by ChadKillingsworth** [comment]
To reword then:

```
namespace "{0}" is required in module {2} but provided in module {1}. Is module {2} missing a dependency on module {1}?
```

**Comment by concavelenz** [comment]
That sounds good to me.  PR?

**Comment by MatrixFrog** [comment]
We already got a PR internally (cl/100422217) but it seems to have stalled.

**Comment by dbarnett** [comment]
I can drop cl/100422217 and follow #1100. Unless it's better having it submitted internally first and synced to github.

**Comment by ChadKillingsworth** [comment]
It makes no difference to me ... Do what's easiest.
